Apply inline

caseyg opened the pull request:
In preparation for the Spring 15 Call for students, I've been working on improving our application flow in a branch called `apply-inline` and wanted to share my progress here. 

## Nav

- Top navigation now links to `Participate` which is information about the call, and `Apply` which is the actual form. Bootstrap has a cool little highlighted label where I put the date, but we could also do a countdown.

![screen shot 2014-12-31 at 8 08 41 pm](https://cloud.githubusercontent.com/assets/462020/5591426/d739785e-9128-11e4-9c03-d373322ce58f.png)

## Participate

1. Short "Open Calls" introduction with buttons for **Apply Now** (jumps to: Apply) or **Read More** (jumps down to full text of Call for Students)
2. Alumni testimonials sidebar, (editable via `/_data/testimonials.yml`) (fixes #61?)
3. "Scenes from School" slideshow (we should add more images)
4. Full text of Spring 2015 Call for Students

![participate school for poetic computation 20141231](https://cloud.githubusercontent.com/assets/462020/5591348/7cbc3be0-9124-11e4-8e4d-4e5692bcdf89.png)

## Apply

1. Migrating from Google Forms (off-site form, spreadsheet backend, free) to Formkeep (on-site form, very usable backend, pay what you wish).
  - With Google Forms we didn't have much control over how the application looked, and answers were stored in a hard to read spreadsheet.
  - Formkeep is simply an endpoint that accepts answers from a form on our site, displays the answers in a much more readable backend, and allows us to search applications and mark spam if necessary.
2. Application is editable via `/_data/application.json`, and is completely customizable.
3. Sidebar shows application/admissions timeline and helpful hints.

![apply school for poetic computation 20141231](https://cloud.githubusercontent.com/assets/462020/5591350/809b6d3a-9124-11e4-94a8-46257ddf3a36.png)

## Thanks

1. After a successful submit, thank the applicant and shower them with virtual confetti. (Looks really smooth in the browser, the GIF just has a low frame rate.)
2. Added prompts to share the Participate page on Facebook and Twitter

![thanks](https://cloud.githubusercontent.com/assets/462020/5591351/88f8d616-9124-11e4-926f-c1cb65eb7783.gif)

## (Optional) Share Prompts

- There are optional, and we can customize what they say/how they look by adding some metadata tags, if anyone has edits.

| Twitter | Facebook |
| --------- | ------------- |
| ![screen shot 2014-12-31 at 7 01 37 pm](https://cloud.githubusercontent.com/assets/462020/5591266/8b09f0f2-911f-11e4-8ead-75f9a25cffe0.png) | ![screen shot 2014-12-31 at 7 01 51 pm](https://cloud.githubusercontent.com/assets/462020/5591268/8e1f18c6-911f-11e4-8cf7-a574620c1451.png) |

## FAQ

- Has a new Table of Contents navigation, which sticks to the screen as you scroll and highlights the answer you're on.

![faq school for poetic computation 20141231](https://cloud.githubusercontent.com/assets/462020/5591352/8ed4deae-9124-11e4-9bcc-846c2eefaaec.png)

## Formkeep Backend

- This is what Formkeep looks like.

![screen shot 2014-12-31 at 6 58 21 pm](https://cloud.githubusercontent.com/assets/462020/5591257/4e894a38-911f-11e4-8cc9-95e66249a978.png)

- - -

## Missing Pieces

I'm excited about all this! Hope you are too.

Here's what's left to do:

- Add in new FAQ content, once finalized (via `/_faqs/section.md`)
- Final application questions (via `/_data/application.json`)
- Final application schedule  (via `/layouts/apply.html`)
- Any design/flow tweaks you might think of
- Internal testing (it worked for me, but would be great to kick the tires.)

caseyg commented:
@tchoi8 is going to speak with Allison tomorrow, so they can give me some feedback on this.

ofZach commented:
cool!  looks good !!  I do feel like testimonials could be it's own page since it makes sense year round.  I feel like we have more than 7, let me dig them up.   we can also get some more from fall 14 folks.

ofZach commented:
also, I think we should do some font work to not have apercu all over the body pages, etc -- it feels like there's some font love needed, things like this seem a bit off to me: 

![screen shot 2015-01-01 at 10 31 46 am](https://cloud.githubusercontent.com/assets/142897/5592524/78f8044e-91a1-11e4-97e9-b7467438f25e.png) 

it's def. not crucial and can come later, just want to mention that finding a good body font and using it all 
throughout the body in different weights, sizes etc makes sense.  I kind of don't like apercu that much through the body, it looks good on the nav / logo to me.

caseyg commented:
Sweet! 

Yeah the type is definitely sloppy, because we still haven't resolved #70 (adding a new body font). Will see if I can do some trial-and-error with the humanist sans-serifs on Fontdeck before we launch this.

caseyg commented:
Just pushed content updates. Still to do:

- Pick a body typeface because bold text will show up weird until we do.
  - Help text type on application needs love.
- Update slideshow with a better breadth of images.
- Move testimonials to their own page?
- Move Call FAQs to FAQ page?
- Create an email alias for whoever wants to be notified of new applications (right now it's just casey@sfpc.io)

ofZach commented:
FAQs have a bit of duplicate content in them, should be edited a bit...

caseyg commented:
@ofZach Agreed, was thinking it'd be a good idea to move any non-time-sensitive Q&A-style content from the /Participate page to the /FAQ page, and then just link to the relevant FAQ questions at the bottom of the /Participate page. It would shorten the call and consolidate duplicate content.

ofZach commented:
looking good, can we drop the alumni <3 sfpc but leave the quotes?

caseyg commented:
Think I've addressed most points in my todo list now https://github.com/SFPC/sfpc.github.io/pull/83#issuecomment-68970548

The only things that're left are not bugs, just enhancements:

- Update slideshow with a better breadth of images.
- Create an email alias for whoever wants to be notified of new applications (right now it's just casey@sfpc.io)

ofZach commented:
I am not super happy with the font change, I put alot of work into getting the type to look good....  Can we revert the changes to the body font please?

ofZach commented:
(alternatively, let me find another font that can work,  helvetica is making my eyes burn)

ofZach commented:
also, did we loose apercu in the nav ?   I think there was some regression here on typography....  this is new vs old

![screen shot 2015-01-07 at 2 10 08 pm](https://cloud.githubusercontent.com/assets/142897/5651618/fcee68ea-9676-11e4-9af7-977bcb68da6f.png)

caseyg commented:
Sometimes the hosted fonts don't load right locally. Pretty sure the nav is still aperçu in the CSS. Try localhost rather than 127.0.0.1?

Would love to move forward with new fonts rather than revert. I feel like you've got a very specific vision for the type, so it’d be super helpful for me to have you make a choice. 

See my comment on https://github.com/SFPC/sfpc.github.io/commit/1aa8f65fb0f910c2cc4e4f7f1b7146ccb618cb12 about removing some hacky font CSS. 

Bootstrap has a type scale that we can customize via http://getbootstrap.com and it adjusts everything in a more proportional way using equations.

ofZach commented:
I did a ton of work on fonts and getting the design right -- either we make it look as good as the old one or we revert, please.  I don't like seeing my hard work trashed.
